src/main/java/mate/project/validation/FieldMatch.java

```diff
@@ -0,0 +1,20 @@
+package mate.project.validation;
+
+import jakarta.validation.Constraint;
+import jakarta.validation.Payload;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Constraint(validatedBy = FieldMatchValidator.class)
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface FieldMatch {
+    String message() default "Fields do not match";
```

Comment:
The default error message 'Fields do not match' is quite generic. It might be more helpful to users if the message specified which fields did not match.